src/main/java/io/cdap/plugin/gcp/bigquery/actions/BigQueryDropDataset.java

```
@@ -0,0 +1,121 @@
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name(BigQueryDropDataset.NAME)
+@Description(BigQueryDropDataset.DESC)
+public class BigQueryDropDataset extends Action {
+  public static final String NAME = "BigQueryDropDataset";
+  public static final String DESC = "Drops a Cloud BigQuery Dataset";
+  private static final String RECORDS_OUT = "records.out";
+  private Config config;
+
+  /**
+   * Drops a dataset.
+   *
+   * @param context a <code>ActionContext</code> instance during runtime.
+   * @throws Exception thrown when delete of dataset fails.
+   */
+  @Override
+  public void run(ActionContext context) throws Exception {
+    BigQuery bq = BigQueryUtil.getBigQuery(config.getServiceAccountFilePath(), config.getProject());
+    try {
+      Boolean deleted = bq.delete(config.getDatasetId());
+      if (deleted) {
+        // the dataset was deleted
+        context.getMetrics().gauge(RECORDS_OUT, 1);
+      } else {
+        // the dataset set was not found.
+        if (config.failIfNotFound()) {
+          throw new NotFoundException(
+            String.format("Failed to delete dataset '%s'. Failing pipeline as configured by user.",
+                          config.getDatasetName())
+          );
+        }
+      }
+    } catch (BigQueryException e) {
+      // the dataset delete had an issue.
+      if (config.failIfNotFound()) {
+        throw new NotFoundException(
+          String.format("User configured pipeline to fail if dataset '%s' was not found. Reason : %s",
+                        config.getDatasetName(), e.getMessage())
+        );
+      }
+    }
+  }
+
+  /**
+   * This class <code>Config</code> provides the configuration for deleting a dataset from a project.
+   */
+  private static final class Config extends GCPConfig {
+    @Name("dataset")
+    @Description("Dataset to be drop from the project")
+    @Macro
+    private String dataset;
+
+    @Name("not-found")
+    @Description("Choice to fail job if BigQuery dataset is not found.")
```

Whether the pipeline should fail if the dataset is not found.